bridge-stp.in

```diff
@@ -212,7 +212,7 @@ case "$action" in
             # Add bridge to mstpd and configure.
             echo "Adding/configuring bridge '$bridge' ..."
             "$mstpctl" addbridge "$bridge" || continue
-            if [ -n "$config_cmd" ]; then
+            if [ -x "$config_cmd" ]; then
```

Comment:
well, we can take this in private, if you want, and i can share some tips

you can usually use the kernel + adapt the switch driver to work with mstpd;
example, for the b53 switch family, you can use something like this:
https://lists.openwrt.org/pipermail/openwrt-devel/2015-February/031542.html
to spawn interfaces for each port, and then the kernel + mstpd can function